port/threads.c

```diff
@@ -96,9 +95,9 @@ static void thread_waittid_thr(void *arg)
 			free(data->stack);
 			free(data);
 
-			mutexLock(global.join_lock);
-			condSignal(global.join_cond);
-			mutexUnlock(global.join_lock);
+			mutexLock(global.lock);
```

Comment:
Is it sensible to free the lock in line 92 just to reacquire it here?

Also: `mutex_unlock` could probably be moved outside of the if-else?